### PR TITLE
[flang] Add genEval to the AbstractConverter

### PR DESCRIPTION
**Pull request by clementval**
There was some discussion on discourse[1] about allowing call to FIR generation functions from other part of lowering belonging to OpenMP.

This solution exposes a simple `genEval` member function on the `AbstractConverter` so that IR generation for PFT Evaluation objects can be called from lowering outside of the FirConverter but not exposing it.

[1] https://discourse.llvm.org/t/openmp-lowering-from-pft-to-fir/75263

### CODE REVIEW

**Review by jeanPerier — APPROVED** [review]
Thanks, I favor this solution since it has a much smaller footprint and only exposes the core utility.

**Review by kparzysz — APPROVED** [review]

File: flang/lib/Lower/Bridge.cpp [hunk 1]
```diff
@@ -839,6 +839,11 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     }
   }
 
+  void genEval(Fortran::lower::pft::Evaluation &eval,
+               bool unstructuredContext) override {
```
Comment (kparzysz) [inline flang/lib/Lower/Bridge.cpp:843]: You can add `final` to it.

**Comment by kparzysz** [comment]
We also need access to the local symbol table.

File: flang/lib/Lower/Bridge.cpp (hunk 1 above)
Comment (clementval) [inline flang/lib/Lower/Bridge.cpp:843]:
```suggestion
               bool unstructuredContext) override final {
```

**Comment by clementval** [comment]
> We also need access to the local symbol table.

What for? It was not part of the discussion we had on discourse? What is the motivation here now that we know more abstraction in the dialect is the right direction to take?

By the way there is no need to expose the local symbol table on the `AbstractConverter`. You can pass it directly to the `genOpenMPConstruct` if needed.

**Review by kparzysz — APPROVED** [review]

**Pull request merged**